cloudbridge/cloud/providers/azure/provider.py

```diff
@@ -113,25 +115,29 @@ def _initialize(self):
         """
         try:
             self._azure_client.get_resource_group(self.resource_group)
-        except CloudError:
-            resource_group_params = {'location': self.region_name}
-            self._azure_client.create_resource_group(self.resource_group,
-                                                     resource_group_params)
-        # Create a storage account. To prevent a race condition, try
-        # to get or create at least twice
-        self._get_or_create_storage_account()
-
-    @tenacity.retry(stop=tenacity.stop_after_attempt(2), reraise=True)
-    def _get_or_create_storage_account(self):
-        try:
-            return self._azure_client.get_storage_account(self.storage_account)
-        except CloudError:
-            storage_account_params = {
-                'sku': {
-                    'name': 'Standard_LRS'
-                },
-                'kind': 'storage',
-                'location': self.region_name,
-            }
-            self._azure_client.create_storage_account(self.storage_account,
-                                                      storage_account_params)
+
+        except CloudError as cloud_error:
+            if "not be found" in cloud_error.message:
+                resource_group_params = {'location': self.region_name}
+
+                try:
+                    self._azure_client.\
```

Comment:
Do we have a race condition identical to the get_or_create_storage situation here? Wrap with tenacity perhaps?

Reply:
We haven't had that happen so far. Do you think we should add it preventively?